src/System.Memory/src/System/Buffers/Text/Base64.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    public static partial class Base64
+    {
+        private static TVector ReadVector<TVector>(ReadOnlySpan<sbyte> data)
```

Comment:
Is used to read a vector from static data, as this safes a few instructions and some memory accesses, like:
```diff
-       48BA800B00240D7F0000 mov      rdx, 0x7F0D24000B80
-       488B0A               mov      rcx, gword ptr [rdx]
-       C5FD104108           vmovupd  ymm0, ymmword ptr[rcx+8]
+       48BA2BEC8871207F0000 mov      rdx, 0x7F207188EC2B
+       C5FD1002             vmovupd  ymm0, ymmword ptr[rdx]

```

Reply:
Does this change with https://github.com/dotnet/coreclr/pull/22944 ?

Reply:
This seems unrelated to me. 

Here it is reading a vector from static data. 
Previously it was assigned from `static readonly Vector256<sbyte>`, now it's read from a ROSpan (i.e. directly from the data).

Reply:
Right, but if that PR eliminates the extra `mov` when loading a vector, this change isn't necessary.  I haven't checked whether it applies when reloading a static vector or if it's just when using `Avx.LoadVector256` explicitly.  In any case, it seems like the JIT should be fixed (if that fix isn't it already) rather than worked around.

Reply:
With the ROSpan the vector is read from the `.text`-section from the assembly, no heap memory and copying is used.
If the vector is statically initialized, the data has to be copyed to the heap (once), and the vector gets read from there. This change avoids this. Granted, the benefit is minimal.